src/main.rs

```diff
@@ -549,6 +550,29 @@ async fn consume_messenger_events(
     Ok(())
 }
 
+// CurrentPeers keeps up-to-date with all of the peers we're connected and disconnected to using a map.
+struct CurrentPeers {
+    map: Mutex<HashMap<PublicKey, bool>>,
```

Comment:
Why do we need a mutex here? If we're only accessing in `consume_messenger_events` then there's no need for the overhead of locking - can we get away with a regular HashMap?

Reply:
Oh d'oh yeah you're right. I was thinking it was called in two threads but it's not. With the mutex and locking removed, the CurrentPeers struct/methods are less necessary. But I still like the peer_connected/peer_disconnected method calls for readability's sake so I'm going to keep it as is if sounds good to you.